### DISCUSSION TITLE
Moving source ZFS without re-replicating

### DISCUSSION THREAD

**Discussion by Malvineous** [zrepl/zrepl#850] [Q&A] [answered]
Hi all,

I'm moving my ZFS datasets onto a new pool, as I have new disks and want to start with a fresh pool configured slightly differently to the old pool.

I used `zfs send` to get the datasets onto the new pool, and I'm now ready to update the mount paths and reboot, to do the cutover make the new pool live.

However, I am not sure how to get zrepl to pick up replication from the new pool and keep updating the existing target data sets?  I think if I just continue as is, zrepl won't know where it is up to and I will have to wipe the target pool and re-replicate everything.

Is it possible to copy something across from the old pool to the new pool, such that zrepl can continue replication where it left off?

I am guessing it is not as simple as copying the latest snapshot across to the new pool with `zfs send`, so any tips would be appreciated!

**Comment by problame** [comment, marked as answer]
If
1. the dataset names on the new pool are identical (including pool name) and
1. the most recent snapshot on the receiving side is present on the sending side,

then replication from the new pool will resume as usual.

**Reply by Malvineous** [reply]
Interesting!  In my case the pool name is changing, which you have made me realise means the replication target dataset will have the old pool's name in it.

Is it possible to `zfs rename` the replicated dataset to match the new pool's name, or is there other state information stored somewhere by zrepl that means a simple rename isn't enough?

**Reply by problame** [reply]
Yeah, zfs rename while zrepl is stopped should do the trick.
